src/main/python/SearchWidget.py

```diff
@@ -0,0 +1,168 @@
+
+        self.finished.emit()
+        self._is_running = False
+
+    def _build_table_text(self, entryText):
+        match_positions = [m.start() for m in re.finditer(self._searchText, entryText, re.IGNORECASE)]
+
+        if len(match_positions) == 0:
+            return ""
+
+        pos = match_positions[0]
+
+        end = pos + 78
+        if end >= len(entryText):
+            table_text = entryText[pos:]
+        else:
+            table_text = entryText[pos:end] + "..."
+
+        if len(match_positions) > 1:
+            table_text = table_text + "(and " + str(len(match_positions)-1) + " more)"
+
+        return table_text
+
+
+class SearchWidget(QtWidgets.QWidget):
+    def __init__(self, edit_pane: EditPane, web_view: WebView):
+        super().__init__()
+
+        self.edit_pane = edit_pane
+        self.web_view = web_view
+        self.worker = None
+
+        self.searchText = QtWidgets.QLineEdit(self)
+        self.searchText.returnPressed.connect(self.search_clicked)
+
+        self.searchButton = QtWidgets.QPushButton("Search", self)
+        self.searchButton.clicked.connect(self.search_clicked)
+
+        toolbar = QtWidgets.QToolBar(self)
+        toolbar.addWidget(self.searchText)
+        toolbar.addWidget(self.searchButton)
+
+        self.progress_bar = QtWidgets.QProgressBar()
+        self.progress_bar.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
+        self.progress_bar.setVisible(False)
+
+        self.table = QtWidgets.QTableWidget(self)
+        self.table.setColumnCount(2)
+        self.table.setHorizontalHeaderLabels(["Date", "Text"])
+        self.table.resizeRowsToContents()
+        self.table.verticalHeader().setVisible(False)
+        self.table.setSelectionBehavior(QtWidgets.QTableView.SelectionBehavior.SelectRows)
+        self.table.setSelectionMode(self.table.SelectionMode.SingleSelection)
+        self.table.setEditTriggers(QtWidgets.QTableWidget.EditTrigger.NoEditTriggers)
+        self.table.itemSelectionChanged.connect(self.on_row_selected)
+        self.table.horizontalHeader().setSortIndicatorShown(True)
+        self.table.setSortingEnabled(True)
+        self.table.setSizeAdjustPolicy(QtWidgets.QAbstractScrollArea.SizeAdjustPolicy.AdjustToContentsOnFirstShow)
+        self.table.horizontalHeader().setStretchLastSection(True)
+
+        layout = QtWidgets.QVBoxLayout()
+        layout.addWidget(toolbar)
+        layout.addWidget(self.progress_bar)
+        layout.addWidget(self.table)
+        self.setLayout(layout)
+
+
+    def search_clicked(self):
+        if self.worker and self.worker.is_running():
+            self.worker.interrupt()
+            return
+
+        self.progress_bar.setValue(0)
+        self.progress_bar.setVisible(True)
+
+        if self.searchText.text() == "":
+            return
+
+        self.table.clearContents()
+        self.table.setRowCount(0)
+
+        # Run search in a background thread
+        self.thread = QtCore.QThread()

```

Comment:
When a long search takes place and you spam click the search/stop button there is a chance to get the error `QThread: Destroyed while thread is still running`

I believe this is due to a new thread being declared each time the button is clicked when in search mode. I think this may be because of an access attempt to the old thread after the new thread has been created and taken its place. 

I have not had the same issue if the line `self.thread = QtCore.QThread()` is placed in `SearchWidget.__init__` this would mean a single thread is used throughout meaning no incorrect access attempts.

I am not experienced in how QThread works so if this fix might lead to further errors let me know

Reply:
Oh no, as soon as multithreading comes into play, things get complicated.

Even with some sleeps to extend search time I could not reproduce the error. I think the checking the is_running flag was the problem as the thread might has not completely terminated even if it is False. Furthermore, creating a single thread instance on __init__ makes sense. I also fixed an error when closing the dialog while the search thread is still running.

So I think everything should be ok now, but please test critically!

Reply:
I have tested for the previous issue and it no longer occurs. Good fix.